src/Microsoft.ML.AutoML/Tuner/CostFrugalTuner.cs

```diff
@@ -13,80 +13,75 @@ internal class CostFrugalTuner : ITuner
 
-        public CostFrugalTuner(AutoMLExperiment.AutoMLExperimentSettings settings, IMetricManager manager)
-            : this(settings.SearchSpace, settings.SearchSpace.SampleFromFeatureSpace(settings.SearchSpace.Default), !manager.IsMaximize)
+        public CostFrugalTuner(AutoMLExperiment.AutoMLExperimentSettings settings)
+            : this(settings.SearchSpace, settings.SearchSpace.SampleFromFeatureSpace(settings.SearchSpace.Default))
         {
         }
 
-        public CostFrugalTuner(SearchSpace.SearchSpace searchSpace, Parameter initValue = null, bool minimizeMode = true)
+        public CostFrugalTuner(SearchSpace.SearchSpace searchSpace, Parameter initValue = null)
         {
             _searchSpace = searchSpace;
-            _minimize = minimizeMode;
-
             _localSearch = new Flow2(searchSpace, initValue, true);
             _currentThreadId = 0;
             _lsBoundMin = _searchSpace.MappingToFeatureSpace(initValue);
             _lsBoundMax = _searchSpace.MappingToFeatureSpace(initValue);
             _initUsed = false;
-            _bestMetric = double.MaxValue;
+            _bestLoss = double.MaxValue;
         }
 
         public Parameter Propose(TrialSettings settings)
         {
             var trialId = settings.TrialId;
+            Parameter param;
             if (_initUsed)
             {
                 var searchThread = _searchThreadPool[_currentThreadId];
-                _configs[trialId] = _searchSpace.MappingToFeatureSpace(searchThread.Suggest(trialId));
+                param = searchThread.Suggest(trialId);
                 _trialProposedBy[trialId] = _currentThreadId;
             }
             else
             {
-                _configs[trialId] = CreateInitConfigFromAdmissibleRegion();
+                param = _searchSpace.SampleFromFeatureSpace(CreateInitConfigFromAdmissibleRegion());
                 _trialProposedBy[trialId] = _currentThreadId;
             }
 
-            var param = _configs[trialId];
-            return _searchSpace.SampleFromFeatureSpace(param);
+            return param;
         }
 
         public Parameter BestConfig { get; set; }
 
         public void Update(TrialResult result)
         {
             var trialId = result.TrialSettings.TrialId;
-            var metric = result.Metric;
-            metric = _minimize ? metric : -metric;
-            if (metric < _bestMetric)
+            var parameter = result.TrialSettings.Parameter;
+            var loss = result.Loss;
+            if (loss < _bestLoss)
             {
-                BestConfig = _searchSpace.SampleFromFeatureSpace(_configs[trialId]);
-                _bestMetric = metric;
+                BestConfig = parameter;
+                _bestLoss = loss;
             }
 
             var cost = result.DurationInMilliseconds;
-            int threadId = _trialProposedBy[trialId];
+            int threadId = _trialProposedBy.ContainsKey(trialId) ? _trialProposedBy[trialId] : _currentThreadId;
```

Comment:
>trialProposedBy.ContainsKey(trialId) ? _trialProposedBy[trialId] : _currentThreadId; [](http://example.com/codeflow?start=28&length=84)

May be better to use `TryGetValue` here to reduce the double search when `trialId` value existed in the dictionary.